File: js210/easy5/double_char2.js
"use strict";

function isConsonant(char) {
  if (char.match(/[^a-zA-Z]/)) {
    return false;
  }

  if (char.match(/[aeiou]/i)) {
    return false;
  }

  return true;
}

function doubleConsonants(str) {
  let doubledLetters = '';

  for (let idx = 0; idx < str.length; idx += 1) {
    if (isConsonant(str[idx])) {
      doubledLetters += (str[idx] + str[idx]);
    } else {
      doubledLetters += str[idx];
    }
  }

  return doubledLetters;
}

console.log(doubleConsonants('String'))
console.log(doubleConsonants('Hello-World!'))
console.log(doubleConsonants('July 4th'))
console.log(doubleConsonants(''))

File: rb101_rb109/easy8/double_char2.rb
=begin
input: a string
output: returns a new string, only every consonant is doubled
=end

CONSONANTS = %w(b c d f g h j k l m n p q r s t v w x y z)

def double_consonants(str)
  result = ""
  str.each_char do |char|
    if CONSONANTS.include?(char.downcase)
      result << char * 2
    else
      result << char
    end
  end
  
  result
end


puts double_consonants("String")
puts double_consonants("Hello-World!")
puts double_consonants("July 14th")
p double_consonants("")